examples/fluids/navierstokes.c

```diff
@@ -312,18 +329,27 @@ int main(int argc, char **argv) {
   CeedQFunctionDestroy(&ceed_data->qf_apply_inflow_jacobian);
   CeedQFunctionDestroy(&ceed_data->qf_apply_freestream);
   CeedQFunctionDestroy(&ceed_data->qf_apply_freestream_jacobian);
+  CeedQFunctionDestroy(&ceed_data->sp_stats.qf_stats_collect);
+  CeedQFunctionDestroy(&ceed_data->sp_stats.qf_stats_proj);
 
   // -- Bases
   CeedBasisDestroy(&ceed_data->basis_q);
   CeedBasisDestroy(&ceed_data->basis_x);
   CeedBasisDestroy(&ceed_data->basis_xc);
   CeedBasisDestroy(&ceed_data->basis_q_sur);
   CeedBasisDestroy(&ceed_data->basis_x_sur);
+  CeedBasisDestroy(&ceed_data->sp_stats.basis_q);
+  CeedBasisDestroy(&ceed_data->sp_stats.basis_x);
 
   // -- Restrictions
-  CeedElemRestrictionDestroy(&ceed_data->elem_restr_q);
-  CeedElemRestrictionDestroy(&ceed_data->elem_restr_x);
-  CeedElemRestrictionDestroy(&ceed_data->elem_restr_qd_i);
+  CeedElemRestrictionDestroy(&ceed_data->er_q);
```

Comment:
Not to bikeshed too much, but this is the opposite of our usual convention to favor words in variable names instead of short initalizations that need decoder rings

Reply:
I have been trying to push towards more explicit variable names because 1) its really cheap to buy extra characters to put in variable names now that we don't have punch cards and 2) its my hope that clearer names make it easier to onboard new students. 

I think some of our variable names are a bit too long, especially for the element restrictions, but I try to find a balance between brevity and clarity.

Reply:
This is fair. And I did start with keeping the `elem_restr_*` syntax, but then ran into things like `ceed_data->sp_stats.elem_restr_parent_colloc`, which is honestly just as difficult to read.

I initially limited this to just the new statistics one (for which I really did need `{parent,child}_colloc` to accurately describe what it was), but figured consistency was better.

My thought for `er_*` was just that we've already done `CeedQFunction -> qf`, `CeedOperator -> op`, so `CeedElemRestriction -> er` seems fairly natural. 

My other, more broad thought is that the variable name "real estate" shouldn't have a lot go to describing the variable type, leaving more for actually describing non-redundant information.

Reply:
I've been trying to use local variables to get around some of this issue of a long name for a member of a struct which is a member of a struct. I believe the compiler optimizes out the stack allocation. 

I don't love putting the type in the variable either but I haven't come up with a good way around it yet. I usually do it because I have op_mass and qf_mass or elem_restr_u, basis_u, and u_loc/U_loc.

Reply:
At some point, we'll remove these things and just have a list/hash of all known qfunctions, then from the input file, match those into `(label, value, qfunction)` for each desired operator, then create the composite operator by turning each of those list entries into a sub-operator. (And this mechanism can largely migrate into PETSc DM.) With this more generic mechanism, we'll be able to ask for a `MatShell` or other operator instead of creating it ourselves. Anyway, I'm not concerned about the length of the names in the struct, just that they follow the same pattern and are uniquely grep'able.

Reply:
Related but not for this PR - I made a few passes through Ratel to reduce the amount of data carried around in these structs.  That removed a lot of unneeded code. A pass or two for that in the fluids code here might be a good idea. That would help with some of this chain of long names, as a nice added benefit.

Reply:
I'm not sure. I'm good with either:
1) Leaving it as it is (`er_q`, etc)
2) Making the new naming scheme more descriptive 
   - Could be`er_q` -> `er_q_vol` to contrast the `*_sur` used else where in the code
   - Could also change `q` to something else, like `state`. That'd require changing a lot of the other variables names throughout the code to be consistent.
3) Changing back to `elem_restr_*`

Reply:
> I don't love putting the type in the variable either but I haven't come up with a good way around it yet. I usually do it because I have op_mass and qf_mass or elem_restr_u, basis_u, and u_loc/U_loc.

I guess my point was less about whether I like the variable type prefix, and more about how much space to dedicate to it. I think having the mix of type-prefix, application-suffix makes for a really useful pattern for understanding the code. I think with `er_*` being used often in that pattern, it becomes straight-forward to remember, similar to `qf`, and `op`.

Reply:
Side note: What is the `*_i` supposed to mean for some of the qdata element restriction. (like `elem_restr_qd_i`)?

Reply:
The _i is actually a legacy convention from when these were called 'identity' restrictions. I've stripped this from Ratel but haven't made a pass over here to do so. We really should, but I haven't made the time. (sort of like how we should reduce the data carried around in the ceed_data struct in a follow up PR, probably need issues for both)

I prefer words in variable names - that's what's in the style guide. The 'elem_restr' prefix, or the op/qf ones don't perfectly match this convention but that's what's used in the rest of the repo. I'd keep the same usage as in the rest of the repo and revisit this when this code moves into another repo or make a separate PR to change the convention in this entire repo all at once. That way we don't mix together functional and purely style issues here.